src/main/java/com/coveo/nashorn_modules/Module.java

```diff
@@ -89,17 +110,20 @@ public Object require(String module) throws ScriptException {
     assert found != null;
     children.add(found.module);
 
+    //remove successful resolves from the refCache
+    refCache.remove(requestedFullPath);
+
     return found.exports;
   }
 
-  private Module searchForModuleInNodeModules(Folder from, String[] folderParts, String filename)
+  private Module searchForModuleInNodeModules(Folder resolvedFolder, String[] folderParts, String filename)
       throws ScriptException {
-    Folder current = from;
+    Folder current = resolvedFolder;
```

Comment:
Yeah, I had to move that up a little because we needed resolved full paths for the refcache.